packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useMemo, useState } from "react"
+import Nestable from "react-nestable"
+import { dropRight, get, flatMap } from "lodash"
+
+import "react-nestable/dist/styles/index.css"
+import "../styles/product-categories.css"
+
+import { ProductCategory } from "@medusajs/medusa"
+import { adminProductCategoryKeys, useMedusa } from "medusa-react"
+
+import TriangleMiniIcon from "../../../components/fundamentals/icons/triangle-mini-icon"
+import ProductCategoryListItemDetails from "./product-category-list-item-details"
+import ReorderIcon from "../../../components/fundamentals/icons/reorder-icon"
+import { useQueryClient } from "@tanstack/react-query"
+import useNotification from "../../../hooks/use-notification"
+
+type ProductCategoriesListProps = {
+  categories: ProductCategory[]
+}
+
+/**
+ * Draggable list that renders product categories tree view.
+ */
+function ProductCategoriesList(props: ProductCategoriesListProps) {
+  const { client } = useMedusa()
+  const queryClient = useQueryClient()
+  const notification = useNotification()
+  const [isUpdating, setIsUpdating] = useState(false)
+
+  const { categories } = props
+
+  const onItemDrop = useCallback(
+    async (params: {
+      item: ProductCategory
+      items: ProductCategory[]
+      path: number[]
+    }) => {
+      setIsUpdating(true)
+      let parentId = null
+      const { dragItem, items, targetPath } = params
+      const [rank] = targetPath.slice(-1)
+
+      if (targetPath.length > 1) {
+        const path = dropRight(
+          flatMap(targetPath.slice(0, -1), (item) => [
+            item,
+            "category_children",
+          ])
+        )
+
+        const newParent = get(items, path)
+        parentId = newParent.id
+      }
+
+      try {
+        await client.admin.productCategories.update(dragItem.id, {
+          parent_category_id: parentId,
+          rank,
+        })
```

Comment:
question: Should we add a hook? 🤔

Reply:
@riqwan Are we missing a hook? Think this is the only question hindering us from merging this one :)

Reply:
We have a hook for update but it accepts category id which is unpractical to use in this case

Reply:
Got it, I don't mind leaving it as is, but ideally we would be using hooks for everything.

Reply:
Agree, currently, we use pass ids to hooks directly to update hooks which has its drawback when the id is dynamically determined and not known during the rendering of the hook, maybe we could write update hooks in the future to pass the id to the mutation function that the hook returns so we have flexibility.